packages/scheduler/lib/models/schedules.ts

```diff
@@ -232,3 +232,26 @@ export async function search(
         return Err(new Error(`Error searching schedules: ${stringifyError(err)}`));
     }
 }
+
+export async function hardDeleteOlderThanNDays(db: knex.Knex, days: number): Promise<Result<DbSchedule[]>> {
+    try {
+        const deleted = await db
+            .from<DbSchedule>(SCHEDULES_TABLE)
+            .where(
+                'id',
+                '=',
+                db.raw(`ANY(ARRAY(
+                    SELECT "id"
+                    FROM ${SCHEDULES_TABLE}
+                    WHERE "deleted_at" < NOW() - INTERVAL '${days} days'
+                    ORDER BY "id" ASC
+                    LIMIT 1000
```

Comment:
This logic will definitely be too "simple" because of foreign key. If a schedule executes every 5minutes, it has ~20K tasks in the last 60days, if you have 1000 schedules that's 20M rows. 
Even deleting 20K can be too slow for PG depending on the lock patterns and db size. I would at least lower the limit drastically

Reply:
discussed online but writing here for visibility:
Cascading deletion of tasks could indeed be problematic if we are deleting a lot of schedules at the same time. In practice schedules are marked as deleted very infrequently so very few would be hard deleted at the same time. However to minimize any potential issue we said we'll set the limit to `1`. For a schedule with frequency 5mins that less than 9k rows

Reply:
limit lowered to 1 https://github.com/NangoHQ/nango/pull/2485/commits/ae0a6c78668e2092933275dd09162ab887d81a75 @bodinsamuel